charts/core/src/ChartHeader/ChartHeader.tsx

```diff
@@ -25,6 +26,7 @@ export function ChartHeader({
         <Body weight="regular" baseFontSize={BaseFontSize.Body1}>
           {title}
         </Body>
+        <div>{titleIcon}</div>
```

Comment:
Nit: Since this could be `undefined` we might want to only render the container `div` if `titleIcon` is defined. This just avoids rendering an empty `div` when it's `undefined`.
```suggestion
        {titleIcon && <div>{titleIcon}</div>}
```